### ISSUE TITLE
systemd unit file fails because of modprobe on LXC

### ISSUE THREAD

**Issue by khaefeli** [containerd/cri#878]
If I install the `containerd` service on an unprivileged LXC container, the systemd unit fails, because it tries to bring up the service with `/sbin/modprobe overlay` 

See [here](https://github.com/containerd/cri/blob/master/contrib/systemd-units/containerd.service)

- OS: Debian Stretch 9.5
- Kernel: 4.17
- LXC 2.0.7-2+deb9u2

this e.g. results in a failed Kubernetes init.

```
Aug 09 14:10:12 kube-controller2 modprobe[28618]: modprobe: ERROR: ../libkmod/libkmod.c:514 lookup_builtin_file() could not open builtin file '/lib/modules/4.16.0-0.bpo.2-amd64/modules.builtin.bin'
Aug 09 14:10:12 kube-controller2 modprobe[28618]: modprobe: FATAL: Module overlay not found in directory /lib/modules/4.16.0-0.bpo.2-amd64
Aug 09 14:10:12 kube-controller2 systemd[1]: containerd.service: Control process exited, code=exited status=1
Aug 09 14:10:12 kube-controller2 systemd[1]: Failed to start containerd container runtime.
```

maybe it would be better to check first if the module is loaded? (overall)

for unprivileged container virtualization I'm not sure which solution is the best.
maybe it would be better to document that overlay needs to be loaded, or it should be checked within the containerd process?

**Comment by mprasil** [comment]
FYI @threesquared (and anyone else that wants to use this quick workaround) it's generally not a good idea to edit files in `/lib/systemd/system/` as they will be overwritten by package updates. The better approach is to create systemd drop-in file and just override a specific value. To do that create file in the following path `/etc/systemd/system/containerd.service.d/00-nomodprobe.conf` with contents: 

```ini
[Service]
ExecStartPre=
```

The path is important, actual file name doesn't really matter as long as it has `.conf` extension.

**Comment by Random-Liu** [comment]
This will be fixed in the coming 1.1.5 release. See https://github.com/containerd/containerd/issues/2772

I'll close this for now. Feel free to reopen if anything goes wrong.
